tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+}
+
+/* This modeling is required because otherwise CBMC takes
+too much time if assigned to minimumBufferSizes. */
+typedef struct _responseHandle
+{
+    struct _httpsResponse RespHandle;
+    char data[MAX_DATA_SIZE];
+} _resHandle_t;
+
+typedef struct _requestHandle
+{
+    struct _httpsRequest ReqHandle;
+    char data[MAX_DATA_SIZE];
+} _reqHandle_t;
+
+typedef struct _connectionHandle
+{
+    struct _httpsConnection pConnHandle;
+    char data[MAX_DATA_SIZE];
+} _connHandle_t;
+
+/* Models the third party HTTP Parser. */
+size_t http_parser_execute (http_parser *parser,
+                            const http_parser_settings *settings,
+                            const char *data,
+                            size_t len) {
+  size_t ret;
+  _httpsResponse_t *_httpsResponse = (_httpsResponse_t *)(parser->data);
+  /* nondet_bool is required to get coverage. */
+  _httpsResponse->foundHeaderField = nondet_bool();
+  _httpsResponse->parserState = PARSER_STATE_BODY_COMPLETE;
+  return ret;
+}
+
+IotNetworkError_t IotNetworkInterfaceCreate( void * pConnectionInfo,
+               void * pCredentialInfo,
+               void * pConnection ) {
+    *(char **)pConnection = malloc(1); /* network connection is opaque */
+    IotNetworkError_t error;
+    return error;
+}
+
+size_t IotNetworkInterfaceSend( void * pConnection,
+           const uint8_t * pMessage,
+           size_t messageLength ) {
+    size_t size;
+    __CPROVER_assume(size <= messageLength);
+    return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceClose( void * pConnection ) {
+    IotNetworkError_t error;
+    return error;
+}
+
+size_t IotNetworkInterfaceReceive( void * pConnection,
+        uint8_t * pBuffer,
+        size_t bytesRequested ) {
+    uint8_t byte;
+    __CPROVER_array_copy(pBuffer,&byte);
+
+    size_t size;
+    __CPROVER_assume(size <= bytesRequested);
+    return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceCallback( void * pConnection,
+                 IotNetworkReceiveCallback_t
+                 receiveCallback,
+                 void * pContext ) {
+    IotNetworkError_t error;
+    return error;
+}
+
+IotNetworkError_t IotNetworkInterfaceDestroy( void * pConnection ) {
+  IotNetworkError_t error;
+  return error;
+}
+
+IotNetworkInterface_t IOTNI = {
+  .create = IotNetworkInterfaceCreate,
+  .close = IotNetworkInterfaceClose,
+  .send = IotNetworkInterfaceSend,
+  .receive = IotNetworkInterfaceReceive,
+  .setReceiveCallback = IotNetworkInterfaceCallback,
+  .destroy = IotNetworkInterfaceDestroy
+};
+
+/* Models the Network Interface. */
+IotNetworkInterface_t *newNetworkInterface() {
+  return &IOTNI;
+}
+
+/* Creates a Connection Info and assigns memory accordingly. */
+IotHttpsConnectionInfo_t * newConnectionInfo() {
+  IotHttpsConnectionInfo_t * pConnInfo = safeMalloc(sizeof(IotHttpsConnectionInfo_t));
+  if(pConnInfo) {
+    pConnInfo->pNetworkInterface = newNetworkInterface();
+    /* The +1 is required for coverage */
```

Comment:
Can you say more for these two comments.  Like "The +1 is required to get coverage of error handlers."